### PR TITLE
Split doc strings in 16380 characters string parts (maximum MSVC capa…

### PR DESCRIPTION
**Pull request by diorcety**
Allow to compile with msvc when string length is more than 16380 characters ([MSVC limit](https://docs.microsoft.com/fr-fr/cpp/error-messages/compiler-errors-1/compiler-error-c2026?view=vs-2019))

### CODE REVIEW

File: pythran/cxxgen.py [hunk 1]
```diff
@@ -461,9 +461,14 @@ def __init__(self, name, docstrings, metadata):
...
-                                    .replace('\n', '\\n')
-                                    .replace('\r', '\\r'))
+        return self.splitstring(dedent(doc).replace('"', '\\"')
+                                           .replace('\n', '\\n')
+                                           .replace('\r', '\\r'))
+
+    def splitstring(self, doc):
+        if len(doc) == 0:
+            return '""'
+        return "".join(['"%s"' % s for s in wrap(doc, 16380)])
```
Comment (serge-sans-paille) [inline pythran/cxxgen.py]: An interesting choice. I wonder if we could take an approach that would prevent this magic (uncommented) number, something along

```
return "".join('"{}\\n"'.format(s) for s in doc.split('\\n'))
```

File: pythran/cxxgen.py [hunk 2]
```diff
@@ -29,7 +29,7 @@
 
 from __future__ import division
 
-from textwrap import dedent
+from textwrap import dedent, wrap
```
Comment (serge-sans-paille) [inline pythran/cxxgen.py]: sorry to bother you with that, but it seems wrap is no longer used :-/

File: pythran/cxxgen.py [hunk 3]
```diff
@@ -461,9 +461,14 @@ def __init__(self, name, docstrings, metadata):
...
     def docstring(self, doc):
-        return '"%s"' % (dedent(doc).replace('"', '\\"')
-                                    .replace('\n', '\\n')
-                                    .replace('\r', '\\r'))
+        return self.splitstring(dedent(doc).replace('"', '\\"')
+                                           .replace('\n', '\\n')
+                                           .replace('\r', '\\r'))
+
+    def splitstring(self, doc):
+        if len(doc) == 0:
```
Comment (serge-sans-paille) [inline pythran/cxxgen.py]: could be ``if not doc``

**Comment by serge-sans-paille** [comment]
@diorcety perfect! can youjust squash your commits?

**Comment by diorcety** [comment]
done

**Comment by serge-sans-paille** [comment]
Thanks!

**Pull request merged**